python/lsst/ip/isr/calibType.py

```diff
@@ -527,7 +527,8 @@ def readFits(cls, filename, **kwargs):
             for k, v in table.meta.items():
                 if isinstance(v, fits.card.Undefined):
                     table.meta[k] = None
-
+        primaryHeader = dict(fits.open(filename)[0].header)
```

Comment:
I think this leaks a filehandle. Please use

```python
with fits.open(filename) as hdul:
    primaryHeader = hudl[0].header
```

PS we should consider at some point changing this to using ResourcePath so that butler can read them directly from S3 without the local file. Something that is entirely possible with `ResourcePath.to_fsspec` with `astropy.io.fits`.

Reply:
Hmm.  Are the files in the LFA?  In which case we have to use resources?

Reply:
The files are in LFA, but there's a `ResourcePath.as_local()` in the `obs_lsst` code, so the calib class only needs to handle local files.  That sentence inverts the logic; the calib class never needed to know where the file actually lived prior to dealing with the photodiode, and so the photodiode ingest code hides that from the calib class.